File: result.go
package fun

const (
	successCode = iota
	cellErrorCode
	errorCode
	closeErrorCode
)

type Result[T any] struct {
	Id     string
	Code   *uint16
	Data   *T
	Msg    *string
	Status uint8
}

func success(data any) Result[any] {
	return Result[any]{Data: &data, Status: successCode}
}

func Error(code uint16, msg string) Result[any] {
	return Result[any]{Code: &code, Msg: &msg, Status: errorCode}
}

func callError(msg string) Result[any] {
	return Result[any]{Msg: &msg, Status: cellErrorCode}
}

func closeError(requestId string) Result[any] {
	return Result[any]{Id: requestId, Status: CloseErrorCode}
}

File: fun/dataType/Service/Result.cs
namespace fun.dataType.Service;
public class Result
{
    public string Id { get; set; }
    public ushort? Code{ get; set; }
    public  object Data{ get; set; }
    public ResultStatusEnum Status{ get; set; }
    public static Result Success(object data)
    {
        return new Result
        {
            Data = data,
            Status = ResultStatusEnum.Success
        };
    }

    public static Result Error(ushort code, object data)
    {
        return new Result
        {
            Code = code,
            Data = data,
            Status = ResultStatusEnum.Error
        };
    }

    internal static Result CallError(object data)
    {
        return new Result
        {
            Data = data,
            Status = ResultStatusEnum.CellError
        };
    }
}